packages/compiler/core/helpers/type-name-utils.ts

```diff
@@ -66,12 +73,21 @@ export function getNamespaceFullName(type: Namespace, options?: TypeNameOptions)
   if (filter && !filter(type)) {
     return "";
   }
+  const segments = [];
+  let current: Namespace | undefined = type;
+  while (current && current.name !== "") {
+    segments.unshift(getIdentifierName(current.name, options));
+    current = current.namespace;
+  }
 
-  return `${getNamespacePrefix(type.namespace, options)}${getIdentifierName(type.name, options)}`;
+  return segments.join(".");
```

Comment:
hhm right, I changed that because otherwise with recursiveness it would omit `TypeSpec` from `TypeSpec.Http` and those other libraries.
Will see how I can still make that work

Reply:
I also didn't want to omit `TypeSpec` or TypeSpect.Reflection if you were actually asking for the namespace directly. Only if you are asking for the qualifier.

Reply:
Interesting, I remember some people were surprised when filtering out Cadl and service namespace did Cadl.Rest.Foo -> Rest.Foo and Service.Something.Foo -> Something.Foo. Maybe we should actually change how the filter works, though it is marginally breaking so we should be sure to document.

Reply:
Moved the filter to be inside the while loop so we get the same behavior. Added a new test file that test the `getTypeName` function

Reply:
Kept the omitting of the std qualifier as in the prefix function as otherwise I think we might want a new option or as you pointed out change the filtering logic.